tests/functional/backend/corpora/test_api.py

```diff
@@ -183,4 +233,4 @@ def test_dataset_upload_flow(self):
 
         # Check that the dataset is gone
         res = requests.get(f"{self.api}/dp/v1/datasets/{dataset_uuid}/status", headers=headers)
-        self.assertEqual(res.status_code, requests.codes.unauthorized)
+        self.assertEqual(res.status_code, requests.codes.forbidden)
```

Comment:
nit: should this be 404 not found instead of forbidden, if the dataset is gone?

Reply:
This is the correct value for now. We will probably change it to something else once we return "removed by user" for deleted public collections.

Reply:
Ahh that's right! Thanks Trent 😄